src/DeFuncto.Core/Types/Du.cs

```diff
@@ -70,5 +71,43 @@ Discriminator switch
         [Pure]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static implicit operator Du<T1, T2>(T2 t2) => new(t2);
+
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Unit Iter(Action<T1> ont1, Action<T2> ont2) =>
+            Match(
+                t1 =>
+                {
+                    ont1(t1);
+                    return unit;
+                },
+                t2 =>
+                {
+                    ont2(t2);
+                    return unit;
+                });
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Unit Iter(Action<T1> fSome) =>
+            Iter(fSome, _ => { });
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Unit Iter(Action<T2> fSome) =>
+            Iter(_ => { }, fSome);
+
+        [Pure]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Unit Iter(Func<T1, Unit> ont1, Func<T2, Unit> ont2) =>
```

Comment:
Since introducing `Unit` is meant to disencourage using `Action`, I would preffer that the default implementation would be the one taking `Func<x, Unit>`

Said that, I'm thinking that the single-case iter overloads, that are there as helpers, should push harder that boundary.

What do yo usay about keeping the All-cases iter for both action and fun, but the single cases only with func?

Reply:
@JYCabello  Sounds good

Reply:
@JYCabello  Then we should inverse the logic for the all the cases so that the one that takes the Funcs is the leading function